compiler-rt/lib/asan/asan_errors.cpp

```diff
@@ -600,6 +602,43 @@ static void PrintShadowMemoryForAddress(uptr addr) {
   Printf("%s", str.data());
 }
 
+static void CheckPoisonRecords(uptr addr) {
+  if (!AddrIsInMem(addr))
+    return;
+  uptr shadow_addr = MemToShadow(addr);
+  unsigned char poison_magic = *(reinterpret_cast<u8 *>(shadow_addr));
+
+  if (poison_magic != kAsanUserPoisonedMemoryMagic)
+    return;
+
+  PoisonRecordRingBuffer *PoisonRecord = AcquirePoisonRecords();
+  if (PoisonRecord) {
+    bool FoundMatch = false;
+
+    for (unsigned int i = 0; i < PoisonRecord->size(); i++) {
+      struct PoisonRecord Record = (*PoisonRecord)[i];
+      if (Record.begin <= addr && addr <= Record.end) {
+        FoundMatch = true;
+
+        StackTrace poison_stack = StackDepotGet(Record.stack_id);
```

Comment:
In general it's not expected StackDepotGet will return something.
It can return empty trace and the message below is bogus.

Reply:
Added a check of the StackTrace size

Reply:
> "ERROR: stack depot did not have a matching stack.\n"

I would not consider this an error. Information is just not avalible.
I'd just quietly print nothing,

Reply:
Done